src/emulator/config/src/config.cpp

```diff
@@ -191,6 +150,48 @@ static ExitCode parse(Config &cfg, fs::path load_path, const std::string &root_p
     return Success;
 }
 
+ExitCode serialize_config(Config &cfg, fs::path output_path) {
```

Comment:
Is output a string view or a copy of a string? If it's possible, you should make it a const reference